pypicloud/cache/dynamo.py

```diff
@@ -22,6 +25,78 @@
 LOG = logging.getLogger(__name__)
 
 
+def _path_depth(package):
+    """Count the slashes in path, or return infinity on no path."""
+    path = package.data.get('path')
+    if not path:
+        return float('+inf')
+    return path.count('/')
+
+
+def _decide_between_versions(contender, current):
+    """Decide between packages with the same filename on different paths."""
+
+    if contender.last_modified < current.last_modified:
+        return contender
```

Comment:
D'oh! Yes, that got lost in the refactor. Also adding in a test to clarify that behaviour (since I should have been bitten by that in the tests, not in the code review).

This ordering is a good bit hard to reason for me too. Let's try to wrangle it!

Two files with the same filename but different path is clearly an operator mistake, and has to be dealt with in the least impactful and surprising manner. The two alternatives I see are:
* Using the latest, the operator will see their change take effect (and will not have their suspicion aroused). The users of the package are now being served a new file for the same exact version. This can lead to problems if the new package is not equivalent to the old one.
* Using the eldest, the operator will not see their change take effect, and will (ostensibly) start investigating why. The users of the package are not affected in any way.

If we compare the necessary work to bring reality to a desired state, we would have, respectively:
* All the engineering org that depends on that file is at risk. Someone has to notice there's a problem, the operator has to be notified of their mistake, remove the new package, let everyone affected know, and these people might have to follow up even further (if they published something that contains the package, like an AMI or something). Or, the files are equivalent and it's all good.
* The operator loses a few hours (probably less than that) investigating why their file upload didn't produce the desired change, and realize they need to bump the package instead (or take some other course of action). This happens even if the files are equivalent.

Given those two alternatives, I've went with using the eldest. There's less potential for fallout, and it makes the symptom happen close to its root cause. Considering the fallout might be unbound (and cascading), I think that justifies the price of having wierd goose chases (and the associated frustration) when the files are equivalent.

LMK if you disagree.